src/controller/cve-id.controller/cve-id.controller.js

```diff
@@ -0,0 +1,234 @@
+      if (Array.isArray(req.query['time_created.gt'])) {
+        return res.status(400).json({ message: 'Only one query parameter of \'' + key + '\' is allowed.' })
+      }
+
+      timestamps.dateOperator.push('gt')
+      timestamps.timeCreated.push(req.query['time_created.gt'].replace(/["']/g, ''))
+    } else {
+      return res.status(400).json({ message: 'Invalid query parameter \'' + key + '\'' })
+    }
+  })
+
+  for (let i = 0; i < timestamps.timeCreated.length; i++) {
+    let value = timestamps.timeCreated[i].match(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}$/)
+
+    if (value) {
+      timestamps.timeCreated[i] = new Date(`${value[0]}.000+00:00`)
+    } else {
+      value = timestamps.timeCreated[i].match(/^\d{4}-\d{2}-\d{2}$/)
+
+      if (value) {
+        timestamps.timeCreated[i] = new Date(`${value[0]}T00:00:00.000+00:00`)
+      } else {
+        return res.status(400).json({ message: 'Timestamp query parameter only supports formats "YYYY-MM-DDTHH:mm:ss" or "YYYY-MM-DD"' })
+      }
+    }
+  }
+
+  const query = {
+    owning_cna: req.header('CVE-API-CNA')
+  }
+
+  if (year) {
+    query.cve_year = year
+  }
+
+  if (state) {
+    query['cve.CVE_data_meta.STATE'] = state.toUpperCase()
+  }
+
+  if (timestamps.timeCreated.length > 0) {
+    query['time.created'] = {}
+
+    for (let i = 0; i < timestamps.timeCreated.length; i++) {
+      if (timestamps.dateOperator[i] === 'lt') {
+        query['time.created'].$lt = timestamps.timeCreated[i]
+      } else {
+        query['time.created'].$gt = timestamps.timeCreated[i]
+      }
+    }
+  }
+
+  Cve.find(query)
+    .exec((err, result) => {
+      if (err) {
+        logger.error(err.stack)
+        return res.status(500).send('Internal Server Error')
+      }
+
+      if (!result || result.length === 0) {
+        return res.status(404).json({ message: 'No CVEs were found for the specified query parameters.' })
+      }
+
+      result.forEach((obj) => {
+        const cna = Object.assign({}, obj)._doc
+        delete cna._id
+      })
+
+      logger.info({ message: 'The CVEs were sent to the user.' })
+      return res.status(200).json(result)
+    })
+}
+
+async function modifyCveId (req, res) {
+  const id = req.params.id
+  const cnaShortName = req.header('CVE-API-CNA')
+  let state = null
+  let newCnaShortname = null
+
+  if (req.query.state) {
+    if (req.query.state.toUpperCase() === 'RESERVED') {
+      return res.status(400).json({ message: 'Cannot change the state to RESERVED.' })
+    }
+
+    state = req.query.state.toUpperCase()
+  }
+
+  if (req.query.cna) {
+    newCnaShortname = req.query.cna
+  }
+
+  Cve.findOne()
+    .byCveId(id)
+    .exec(async (err, result) => {
+      if (err) {
+        logger.error(err.stack)
+        return res.status(500).send('Internal Server Error')
+      }
+
+      if (!result) {
+        return res.status(404).json({ message: id + ' not found' })
+      }
+
+      if (cnaShortName !== process.env.SECRETARIAT) {
```

Comment:
This should be at the top. If they aren’t secretariat, they can’t use this endpoint period.